File: VB/RangedMaskSample/MaskElementsRepository.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic

Namespace RangedMaskSample
	Public NotInheritable Class MaskElementsRepository
		Private Shared aCollection As MaskElementCollection
		Private Shared aEditList As List(Of MaskButtonEdit)

		Private Sub New()
		End Sub
		Shared Sub New()
			aCollection = New MaskElementCollection()
			aEditList = New List(Of MaskButtonEdit)()
		End Sub

		Public Shared ReadOnly Property MaskCollection() As MaskElementCollection
			Get
				Return aCollection
			End Get
		End Property

		Public Shared ReadOnly Property MaskButtonEditList() As List(Of MaskButtonEdit)
			Get
				Return aEditList
			End Get
		End Property

		Public Shared Function Add(ByVal aElement As MaskElement) As Integer
			Return aCollection.Add(aElement)
		End Function
	End Class
End Namespace

File: CS/RangedMaskSample/MaskElementsRepository.cs
using System;
using System.Collections.Generic;

namespace RangedMaskSample
{
    public static class MaskElementsRepository
    {
        private static MaskElementCollection aCollection;
        private static List<MaskButtonEdit> aEditList;

        static MaskElementsRepository()
        {
            aCollection = new MaskElementCollection();
            aEditList = new List<MaskButtonEdit>();
        }

        public static MaskElementCollection MaskCollection
        {
            get
            {
                return aCollection;
            }
        }

        public static List<MaskButtonEdit> MaskButtonEditList
        {
            get
            {
                return aEditList;
            }
        }

        public static int Add(MaskElement aElement)
        {
            return aCollection.Add(aElement);
        }
    }
}
